testssl.sh

```diff
@@ -22055,8 +22229,15 @@ determine_optimal_proto() {
           ignore_no_or_lame " Type \"yes\" to proceed and accept false negatives or positives" "yes"
           [[ $? -ne 0 ]] && exit $ERR_CLUELESS
      fi
-
+     actual_proto="$(get_protocol $TMPFILE)"
      tmpfile_handle ${FUNCNAME[0]}.txt
+     # if the actual_proto is TLSv1.2 and TLSv1.3 is available it probably means that the
+     # current openssl version does not support TLSv1.3 so tls_sockets must be used to 
+     # obtain information about the certificate request
```

Comment:
I don't understand this. If $OPENSSL and the server support TLS 1.2, then `sclient_auth()` will correctly determine whether the server sends a CertificateRequest message. It will not fill in `CLIENT_AUTH_SIGALGS_LIST_13` and `CLIENT_AUTH_SIGALGS_CERT_LIST_13`. But then, if $OPENSSL and the server both support TLS 1.3, the code above will not fill in `CLIENT_AUTH_SIGALGS_LIST_12`, and the code below doesn't address that.

Also, this code does nothing in the case that $OPENSSL does not support TLS 1.3 and the server _only_ supports TLS 1.3. Running it seems inappropriate when `$CLIENT_AUTH` is `unknown`.

It seems that this code isn't needed. In the case that the server supports TLS 1.2, but $OPENSSL does not, calls to `tls_sockets()` from within `get_server_certificate()` will already set the necessary variables. So, there is no need for a special call here.